src/include/processor/operator/partitioner.h

```diff
@@ -61,11 +63,19 @@ struct PartitionerSharedState {
     void resetState();
     void merge(std::vector<std::unique_ptr<PartitioningBuffer>> localPartitioningStates);
 
-    storage::InMemChunkedNodeGroupCollection& getPartitionBuffer(common::idx_t partitioningIdx,
-        common::partition_idx_t partitionIdx) const {
+    // Must only be called once for any given parameters
+    std::unique_ptr<storage::InMemChunkedNodeGroupCollection> getPartitionBuffer(
```

Comment:
For what part? It only can be called once for a given parameter pair since it moves the InMemChunkedNodeGroupCollection out of the shared state.
And loadFromDisk is called every time since it doesn't do anything if the data is already in memory.

Reply:
ah sorry, I got it after a second look, but forgot to remove the comment here. mark it as resolved now.